base_layer/core/src/blocks/genesis_block.rs

```diff
@@ -299,127 +367,26 @@ fn get_raw_block(genesis_timestamp: &DateTime<FixedOffset>) -> Block {
-        // if consensus values change, e.g. new faucet or other
-        let block = get_stagenet_genesis_block();
-        check_block(Network::StageNet, &block, 0, 0);
-    }
-
-    #[test]
-    fn nextnet_genesis_sanity_check() {
-        // Note: Generate new data for `pub fn get_nextnet_genesis_block()` and `fn get_stagenet_genesis_block_raw()`
-        // if consensus values change, e.g. new faucet or other
-        let block = get_nextnet_genesis_block();
-        check_block(Network::NextNet, &block, 0, 0);
-    }
+    use crate::consensus::ConsensusManager;
 
     #[test]
-    fn esmeralda_genesis_sanity_check() {
-        // Note: Generate new data for `pub fn get_esmeralda_genesis_block()` and `fn get_esmeralda_genesis_block_raw()`
-        // if consensus values change, e.g. new faucet or other
-        let block = get_esmeralda_genesis_block();
-        check_block(Network::Esmeralda, &block, 4965, 1);
-    }
-
-    #[test]
-    fn igor_genesis_sanity_check() {
-        // Note: Generate new data for `pub fn get_igor_genesis_block()` and `fn get_igor_genesis_block_raw()`
-        // if consensus values change, e.g. new faucet or other
-        let block = get_igor_genesis_block();
-        check_block(Network::Igor, &block, 5526, 1);
-    }
-
-    fn check_block(network: Network, block: &ChainBlock, expected_outputs: usize, expected_kernels: usize) {
-        assert!(block.block().body.inputs().is_empty());
-        assert_eq!(block.block().body.kernels().len(), expected_kernels);
-        assert_eq!(block.block().body.outputs().len(), expected_outputs);
-
-        let factories = CryptoFactories::default();
-        let some_output_is_coinbase = block.block().body.outputs().iter().any(|o| o.is_coinbase());
-        assert!(!some_output_is_coinbase);
-        let outputs = block.block().body.outputs().iter().collect::<Vec<_>>();
-        batch_verify_range_proofs(&factories.range_proof, &outputs).unwrap();
-        // Coinbase and faucet kernel
-        assert_eq!(
-            block.block().body.kernels().len() as u64,
-            block.header().kernel_mmr_size
-        );
-        assert_eq!(
-            block.block().body.outputs().len() as u64,
-            block.header().output_mmr_size
-        );
-
-        for kernel in block.block().body.kernels() {
-            kernel.verify_signature().unwrap();
-        }
-        let some_kernel_contains_coinbase_features = block
-            .block()
-            .body
-            .kernels()
-            .iter()
-            .any(|k| k.features.contains(KernelFeatures::COINBASE_KERNEL));
-        assert!(!some_kernel_contains_coinbase_features);
-
-        // Check MMR
-        let mut kernel_mmr = KernelMmr::new(Vec::new());
-        for k in block.block().body.kernels() {
-            kernel_mmr.push(k.hash().to_vec()).unwrap();
+    fn genesis_block_sanity_check() {
+        for network in &[
+            // TODO: mainnet not yet implemented, uncomment when ready
```

Comment:
```suggestion
```
I have gone through effort to remove all todo's and create issues for them. 
The uncommented Mainnet is fine